### PR TITLE
kafka-multi-cluster

### PR DESCRIPTION
**Pull request by SEQUOIIA**
DON'T MERGE.

### CODE REVIEW

**Comment by devex-sa** [comment]
[//]: # (4744e426-5795-11e9-8647-d663bd873d93)

|| **CodeScene Delta Analysis Results** |
| --- | ---- |
| **Risk** |10|
| **Quality Gates** |Fail|
| **Description** |Be careful with the change -- it degrades the code health.|
| **Warnings** |Degrades in Code Health<br/><ul><li>TwoTopicsWithTheSameNameAndUniqueCluster.cs degrades from a Code Health of 10.0 -> 9.33</li><li>TwoTopicsWithTheSameNameAndUniqueCluster.cs degrades from a Code Health of 10.0 -> 9.33</li></ul>|
| **Code Health Delta Descriptions** |TwoTopicsWithTheSameNameAndUniqueCluster.cs<br/><ul><li> **Degradations:** <ul><li>Code Duplication (functions with similar structure: And_a_topic,When_a_topic_with_the_same_name_but_different_cluster_is_added) - new issue</li></ul></li></ul><br/>TwoTopicsWithTheSameNameAndUniqueCluster.cs<br/><ul><li> **Degradations:** <ul><li>Code Duplication (functions with similar structure: And_a_topic,When_a_topic_with_the_same_name_but_different_cluster_is_added) - new issue</li></ul></li></ul>|
| **New Files with Low Code Health** |9 new files where the code health is above the lower threshold for new code (8.1). 0 new files below the threshold.|

**Comment by devex-sa** [comment]
[//]: # (4744e426-5795-11e9-8647-d663bd873d93)

|| **CodeScene Delta Analysis Results** |
| --- | ---- |
| **Risk** |10|
| **Quality Gates** |Fail|
| **Description** |Be careful with the change -- it degrades the code health.|
| **Warnings** |Degrades in Code Health<br/><ul><li>TwoTopicsWithTheSameNameAndUniqueCluster.cs degrades from a Code Health of 10.0 -> 9.33</li><li>TwoTopicsWithTheSameNameAndUniqueCluster.cs degrades from a Code Health of 10.0 -> 9.33</li></ul>|
| **Code Health Delta Descriptions** |TwoTopicsWithTheSameNameAndUniqueCluster.cs<br/><ul><li> **Degradations:** <ul><li>Code Duplication (functions with similar structure: And_a_topic,When_a_topic_with_the_same_name_but_different_cluster_is_added) - new issue</li></ul></li></ul><br/>TwoTopicsWithTheSameNameAndUniqueCluster.cs<br/><ul><li> **Degradations:** <ul><li>Code Duplication (functions with similar structure: And_a_topic,When_a_topic_with_the_same_name_but_different_cluster_is_added) - new issue</li></ul></li></ul>|
| **New Files with Low Code Health** |9 new files where the code health is above the lower threshold for new code (8.1). 0 new files below the threshold.|

File: src/CapabilityService.WebApi/Features/Kafka/Domain/Events/ClusterCreated.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using DFDS.CapabilityService.WebApi.Domain.Events;
+using DFDS.CapabilityService.WebApi.Features.Kafka.Domain.Models;
+
+namespace DFDS.CapabilityService.WebApi.Features.Kafka.Domain.Events
+{
+	public class ClusterCreated : IDomainEvent
+	{
+		public string Name { get; set; }
```
Comment (ZaradarBH) [inline src/CapabilityService.WebApi/Features/Kafka/Domain/Events/ClusterCreated.cs:9]: protected accessor? use ctor to set member values

File: src/CapabilityService.WebApi/Features/Kafka/Domain/Models/Cluster.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using DFDS.CapabilityService.WebApi.Domain.Models;
+using DFDS.CapabilityService.WebApi.Features.Kafka.Domain.Events;
+
+namespace DFDS.CapabilityService.WebApi.Features.Kafka.Domain.Models
+{
+	public class Cluster : AggregateRoot<Guid>
+	{
+		public string Name { get; set; }
```
Comment (ZaradarBH) [inline src/CapabilityService.WebApi/Features/Kafka/Domain/Models/Cluster.cs:9]: set == protected

File: src/CapabilityService.WebApi/Features/Kafka/Domain/Repositories/IClusterRepository.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using DFDS.CapabilityService.WebApi.Features.Kafka.Domain.Models;
+
+namespace DFDS.CapabilityService.WebApi.Features.Kafka.Domain.Repositories
+{
+	public interface IClusterRepository
+	{
+		Task<IEnumerable<Cluster>> GetAllAsync();
```
Comment (ZaradarBH) [inline src/CapabilityService.WebApi/Features/Kafka/Domain/Repositories/IClusterRepository.cs:10]: maybe add an optional predicate param?

File: src/CapabilityService.WebApi/Features/Kafka/Infrastructure/Api/ClusterController.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
+﻿using System.Threading.Tasks;
+using DFDS.CapabilityService.WebApi.Features.Kafka.Domain.Repositories;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace DFDS.CapabilityService.WebApi.Features.Kafka.Infrastructure.Api
+{
+	[Authorize(AuthenticationSchemes = "AzureADBearer")]
```
Comment (ZaradarBH) [inline src/CapabilityService.WebApi/Features/Kafka/Infrastructure/Api/ClusterController.cs:8]: We need to replace this with MicrosoftIdentityWeb packge

**Pull request closed without merging**